### PR TITLE
jjb: submit-project: also submit ardana packages

### PR DESCRIPTION
**Pull request by jdsn**
* refactoring of the job
* adding ardana package submission to OBS

### CODE REVIEW

File: jenkins/ci.suse.de/openstack-submit-project.yaml [hunk 1]
```diff
@@ -25,90 +25,91 @@
...
-          # avoid copying untested versions
-          lastchange=$($OSC api "source/$COSS/$COMPONENT/_history" | perl -ne 'if(m/<time>(\d+)</) {print $1; exit 0}')
-          if [[ $lastchange > $starttime ]] ; then
-              echo "skipping $COMPONENT because of a change after test started"
-              continue
-          fi
+              # avoid copying untested versions
+              lastchange=$($osc api "source/$coss/$component/_history" | perl -ne 'if(m/<time>(\d+)</) {print $1; exit 0}')
+              if [[ $lastchange -gt $starttime ]] ; then
+                  echo "Skipping $component because due to a change after the test finishe. test time: $starttime :: last change: $lastchange"
```
Comment (rsalevsky) [inline jenkins/ci.suse.de/openstack-submit-project.yaml]: typo finished

Comment (jdsn) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: fixed

File: jenkins/ci.suse.de/openstack-submit-project.yaml [hunk 2]
```diff
@@ -42,43 +42,43 @@
...
           esac
 
-          export BS=IBS
-          OSC_OBS="osc -A https://api.opensuse.org"
-          OSC_IBS="osc -A https://api.suse.de"
-          export OSC="$OSC_IBS"
+          export bs=IBS
+          osc_obs="osc -A https://api.opensuse.org"
+          osc_ibs="osc -A https://api.suse.de"
+          export osc="$osc_ibs"
```
Comment (bmwiedemann) [inline jenkins/ci.suse.de/openstack-submit-project.yaml]: if this is an internal variable - why did you add export in the 1st commit here?

Comment (jdsn) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: good point - it was in the original code (export OSC ; export BS)
will fix it

File: jenkins/ci.suse.de/openstack-submit-project.yaml [hunk 3]
```diff
@@ -25,90 +25,91 @@
...
+              None) echo "No valid 'subproject' provided - defaults for safety reasons to 'None'"
+                    exit 1
+              ;;
+              "-")  echo "Subproject set to '-': skip job, do not fail (eg. for Master project)"
+                    exit 0
+              ;;
           esac
 
+          export bs=IBS
+          osc_obs="osc -A https://api.opensuse.org"
```
Comment (bmwiedemann) [inline jenkins/ci.suse.de/openstack-submit-project.yaml]: 2 unused variables?

Comment (jdsn) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: `bs` is used

Comment (jdsn) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: and the osc_obs variable was used in earlier versions of the script (when the job still was a stand alone shell script)
sure it can be dropped

File: jenkins/ci.suse.de/openstack-submit-project.yaml [hunk 4]
```diff
@@ -25,90 +25,83 @@
...
+              if ! grep -q "<linkinfo" .osc/_files ; then
+                  echo "Error: package $component in $subproject (staging) is not a link to non-staging."
+                  exit 2
+              fi
 
-          grep -q "<linkinfo" .osc/_files || exit 2
+              # avoid copying untested versions
+              lastchange=$($OSC api "source/$COSS/$component/_history" | perl -ne 'if(m/<time>(\d+)</) {print $1; exit 0}')
+              if [[ $lastchange -gt $starttime ]] ; then
+                  echo "Skipping $component because due to a change after the test finished. test time: $starttime :: last change: $lastchange"
```
Comment (bmwiedemann) [inline jenkins/ci.suse.de/openstack-submit-project.yaml]: `because due to` doesnt sound like correct English.
and $starttime is meant to be the time the test started, not `finished` because if we start a test, then something changes and then we finish, we must not submit it. (there will be another testrun later)

Comment (jdsn) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: will fix the wording,
and the starttime is the time that previous job finished - the submit job is following the actual "test job"

File: jenkins/ci.suse.de/openstack-submit-project.yaml [hunk 5]
```diff
@@ -25,90 +25,83 @@
...
+              ;;
+              Devel:Cloud:8)
+                  OSCBUILDDIST=SLE_12_SP3
+                  ECOS=systemsmanagement:crowbar:5.0
+              ;;
+              *) echo "Please add support for the project: $project"
+                  exit 1
           esac
 
+          pkg_filter=`echo ${packagefilter}`
```
Comment (bmwiedemann) [inline jenkins/ci.suse.de/openstack-submit-project.yaml]: why this echo? e.g. it will interpret `*` as shell glob which can have unexpected effects.

Comment (jdsn) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: if it contains repeated white spaces, these will be reduced to one - otherwise the next line will produce wrong regexp syntax

Comment (bmwiedemann) [reply jenkins/ci.suse.de/openstack-submit-project.yaml]: would still be good to not have bash interpret that - e.g. `pkg_filter=$(echo "$pkg_filter" | sed -e 's/  */ /g')`
